File: exemplos/aula_250617/example1.py
try:
    numerador = 10
    denominador = int(input("Digite um número para dividir 10: "))
    resultado = numerador / denominador
    print(f"O resultado é {resultado}")
# Captura uma exceção específica
except ZeroDivisionError:
    print("Erro: Não é possível dividir por zero! 🚫")
# Captura outra exceção específica
except ValueError:
    print("Erro: Você precisa digitar um número inteiro! 🔢")
# Captura qualquer outra exceção (usar com cuidado!)
except Exception as e:
    print(f"Ocorreu um erro inesperado: {e}")
finally:    
    print("Fim do da operação de divisão! 🎉")

File: exemplos/aula_250617/example1.java
import java.util.Scanner;
import java.util.InputMismatchException;

class Divisao {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Digite um número para dividir 10: ");
        try {
            int denominador = scanner.nextInt();
            int resultado = 10 / denominador;
            System.out.println("O resultado é: " + resultado);
        // Captura uma exceção específica
        } catch (ArithmeticException e) {
            System.out.println("Erro: Não é possível dividir por zero! ");
        // Captura outra exceção específica
        } catch (InputMismatchException e) {
            System.out.println("Erro: Você precisa digitar um número inteiro! 🔢");
        // Captura a classe pai de muitas exceções
        } catch (Exception e) {
            System.out.println("Ocorreu um erro inesperado: " + e.getMessage());
        }
        finally {
            scanner.close();
            System.out.println("Fim do da operação de divisão! 🎉");
        }
    }
}